Bottom: a0d620c83a5f202817b4ad05221e95008e34a381
Top:    41e944da9e3a2e6f7aecde72a095aee914508844
Author: Bart Van Assche <bvanassche@acm.org>
Date:   2013-07-28 11:20:54 +0200

[4/4] Declare 'server_port' as an unsigned variable

Change the data type of the 'server_port' variable from signed to
unsigned such that the cast in the fscanf() call can be removed.

Signed-off-by: Bart Van Assche <bvanassche@acm.org>


---

diff --git a/src/acm.c b/src/acm.c
index 49bda48..33379cf 100644
--- a/src/acm.c
+++ b/src/acm.c
@@ -35,6 +35,7 @@
 #endif /* HAVE_CONFIG_H */
 
 #include <stdio.h>
+#include <inttypes.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netdb.h>
@@ -113,7 +114,7 @@ struct acm_msg {
 
 static pthread_mutex_t acm_lock = PTHREAD_MUTEX_INITIALIZER;
 static int sock = -1;
-static short server_port;
+static uint16_t server_port;
 
 struct ib_connect_hdr {
 	uint8_t  cma_version;
@@ -132,7 +133,7 @@ static int ucma_set_server_port(void)
 	FILE *f;
 
 	if ((f = fopen("/var/run/ibacm.port", "r" STREAM_CLOEXEC))) {
-		fscanf(f, "%hu", (unsigned short *) &server_port);
+		fscanf(f, "%" SCNu16, &server_port);
 		fclose(f);
 	}
 	return server_port;
